### PR TITLE
[export] Serialize subgraphs.

### PR DESCRIPTION
**Pull request by zhxchen17**
Differential Revision: D46865179

Deserialization part will be added in a following up PR.

### CODE REVIEW

File: torch/_export/serde/serialize.py [hunk 1]
```diff
@@ -463,11 +485,22 @@ def handle_call_function(self, node: torch.fx.Node):
                 # TODO: create a new tensor_values here, meta might have faketensor info
                 metadata=serialize_metadata(node),
             )
+        elif isinstance(node.target, torch._ops.HigherOrderOperator):
+            assert isinstance(node.meta['val'], FakeTensor), "Only single tensor output is supported for HigherOrderOperator serialization."
+            inputs = [NamedArgument(
+                name="", # TODO(zhxchen17) This is sad, should be improved when HOO has schema arg names.
+                arg=self.serialize_input(arg),
+            ) for arg in node.args]
```
Comment (larryliu0820) [inline torch/_export/serde/serialize.py:493]: `arg` is shadowing `NamedArgument.arg`. Consider changing the variable name.

**Review by larryliu0820 — COMMENTED** [review]
Can you add a test for this?

**Comment by zhxchen17** [comment]
> Can you add a test for this?

I will add a test in the deserialization diff later.

**Review by larryliu0820 — APPROVED** [review]

File: torch/_export/serde/serialize.py [hunk 2]
```diff
@@ -513,18 +549,21 @@ def serialize_inputs(
...
+            isinstance(arg, torch.fx.Node) and arg.name in self.graph_state.sym_bool_values
         )
 
     def serialize_input(self, arg) -> Argument:
         if isinstance(arg, torch.fx.Node):
             if arg.op == "get_attr":
-                return Argument.create(as_tensor=TensorArgument(name=str(arg.target)))
+                assert isinstance(arg.target, str)
+                with self.save_graph_state():
+                    graph = self.serialize_graph(getattr(arg.graph.owning_module, arg.target))
```
[the remark this answers is not in the document]
Comment (zhxchen17) [reply torch/_export/serde/serialize.py:565]: ??

**Pull request closed without merging**